doc/xgboost_on_sqlflow.md

```diff
@@ -0,0 +1,139 @@
+# xgboost on sqlflow
```

Comment:
Capitalize the first letter in the titles.

Reply:
This is a design doc, the title and file name should be like `# Design: ....` so that readers won't think it's a tutorial or manual.